### PR TITLE
ci: Build wasm query engine within nix environement

### PR DESCRIPTION
**Pull request by SevInf**
Ensures that all necessary tooling (in particular, wasm-opt)  is always available for when it is
built. Affected jobs:

- Adapter tests
- Size comparison job
- WASM engine release

Fix https://github.com/prisma/team-orm/issues/752

### CODE REVIEW

File: nix/all-engines.nix
Comment (SevInf) [inline nix/all-engines.nix:156]: More or less copying what `prisma-schema-wasm` already does during publishing.

File: .github/workflows/wasm-size.yml
Comment (SevInf) [inline .github/workflows/wasm-size.yml:31]: `result` and `result-1` are symlinks that `nix build` creates nad they correspond to wasm and wasm.gz packages respectively.

File: nix/shell.nix
Comment (SevInf) [inline nix/shell.nix:22]: Contains wasm-opt

**Comment by jkomyno** [comment]
I've tested this PR here: https://github.com/prisma/prisma-engines/pull/4586#issuecomment-1860901707.

Let's wait for @aqrln's approval, though.

**Review by jkomyno — APPROVED** [review]

File: nix/all-engines.nix
Comment (aqrln) [inline nix/all-engines.nix:119]:
```suggestion
```

**Review by aqrln — APPROVED** [review]

**Pull request merged**
